### PR TITLE
test: Improve flakey tests

### PR DESCRIPTION
**Pull request by ColinKYuen**
### Summary

test: Improve flakey tests

### Description

- Tagging cluster, for maint window to auto delete
- Test ordering. Network > Hard
- Masking session token
- Query for ID rather than use SDK
- Moved test clusterEndpoint to AuroraIntegration

### Additional Reviewers

@karenc-bq @crystall-bitquill @sergiyv-bitquill @matthewh-BQ

### CODE REVIEW

File: src/test/java/testsuite/integration/container/IntegrationTestSuite.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+import org.junit.platform.suite.api.Suite;
+
+@Suite
+@SelectClasses({
+    AuroraMysqlIntegrationTest.class,
+    AuroraMysqlFailoverIntegrationTest.class,
+    AuroraMysqlHardFailureIntegrationTest.class,
+    ReplicationFailoverIntegrationTest.class
+})
+public class IntegrationTestSuite {}
```
Comment (sergiyv-bitquill) [inline src/test/java/testsuite/integration/container/IntegrationTestSuite.java:13]: What's the purpose of this class?

File: src/test/java/testsuite/integration/utility/AuroraTestUtility.java [hunk 2]
```diff
@@ -182,7 +186,8 @@ public String createCluster() throws InterruptedException {
             .withSourceRegion(dbRegion)
             .withEnableIAMDatabaseAuthentication(true)
             .withEngine(dbEngine)
-            .withStorageEncrypted(true);
+            .withStorageEncrypted(true)
+            .withTags(testRunnerTag);
```
Comment (sergiyv-bitquill) [inline src/test/java/testsuite/integration/utility/AuroraTestUtility.java:190]: What is it for?

File: src/test/java/testsuite/integration/container/IntegrationTestSuite.java (hunk 1 above)
Comment (ColinKYuen) [reply src/test/java/testsuite/integration/container/IntegrationTestSuite.java:13]: To run the tests in a specific order. Running the tests with network failure first then running the tests that issue a reboot/failover on the instances.

File: src/test/java/testsuite/integration/utility/AuroraTestUtility.java (hunk 2 above)
Comment (ColinKYuen) [reply src/test/java/testsuite/integration/utility/AuroraTestUtility.java:190]: Setting up for AWS Maintenance Windows to automatically delete the clusters if for some reason, the tests did not delete them. I've noticed it happen twice and this is just another safety net / to catch if it fails.

File: build.gradle.kts [hunk 3]
```diff
@@ -384,10 +385,7 @@ tasks.register<Test>("debug-community-docker") {
 }
 
 tasks.register<Test>("in-container-aurora") {
-    filter.includeTestsMatching("testsuite.integration.container.AuroraMysqlIntegrationTest")
-    filter.includeTestsMatching("testsuite.integration.container.AuroraMysqlFailoverIntegrationTest")
-    filter.includeTestsMatching("testsuite.integration.container.AuroraMysqlHardFailureIntegrationTest")
-    filter.includeTestsMatching("testsuite.integration.container.ReplicationFailoverIntegrationTest")
+    filter.includeTestsMatching("testsuite.integration.container.IntegrationTestSuite")
```
Comment (sergiyv-bitquill) [inline build.gradle.kts:388]: If we want to control order of execution for integration tests then I'd recommend add a comment here and explain how to add more test classes.

**Review by sergiyv-bitquill — APPROVED** [review]

File: build.gradle.kts [hunk 4]
```diff
@@ -221,6 +221,7 @@ dependencies {
     testImplementation("org.junit.platform:junit-platform-commons:1.8.2")
     testImplementation("org.junit.platform:junit-platform-engine:1.8.2")
     testImplementation("org.junit.platform:junit-platform-launcher:1.8.2")
+    testImplementation("org.junit.platform:junit-platform-suite-engine:1.8.2")
```
Comment (karenc-bq) [inline build.gradle.kts:224]: I wonder if there is a dependency that includes all these subpackages.

File: src/test/java/testsuite/integration/container/AuroraMysqlIntegrationBaseTest.java [hunk 5]
```diff
@@ -382,7 +386,6 @@ protected void makeSureInstancesUp(String[] instances, boolean finalCheck) throw
       executorService.submit(() -> {
         while (true) {
           try (final Connection conn = connectToInstance(id + DB_CONN_STR_SUFFIX, MYSQL_PORT, initFailoverDisabledProps())) {
-            conn.close();
```
Comment (karenc-bq) [inline src/test/java/testsuite/integration/container/AuroraMysqlIntegrationBaseTest.java:385]: Why are we removing this close statement?

File: src/test/java/testsuite/integration/container/IntegrationTestSuite.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package testsuite.integration.container;
```
Comment (karenc-bq) [inline src/test/java/testsuite/integration/container/IntegrationTestSuite.java:1]: License

File: src/test/java/testsuite/integration/host/AuroraIntegrationContainerTest.java [hunk 7]
```diff
@@ -113,7 +113,7 @@
   static void setUp() throws SQLException, InterruptedException, UnknownHostException {
     Assertions.assertNotNull(AWS_ACCESS_KEY_ID);
     Assertions.assertNotNull(AWS_SECRET_ACCESS_KEY);
-    Assertions.assertNotNull(AWS_SESSION_TOKEN);
+//    Assertions.assertNotNull(AWS_SESSION_TOKEN);
```
Comment (karenc-bq) [inline src/test/java/testsuite/integration/host/AuroraIntegrationContainerTest.java:116]: Don't forget to uncomment

File: src/test/java/testsuite/integration/utility/ContainerHelper.java [hunk 8]
```diff
@@ -250,6 +250,25 @@ public ToxiproxyContainer createAndStartProxyContainer(
...
+
+    ArrayList<String> mySqlInstances = new ArrayList<>();
+
+    try (final Connection conn = DriverManager.getConnection(connectionUrl, userName, password);
+        final Statement stmt = conn.createStatement()) {
+      // Get instances
+      try (final ResultSet resultSet = stmt.executeQuery(RETRIEVE_TOPOLOGY_SQL)) {
+        while (resultSet.next()) {
+          // Get Instance endpoints
+          final String hostEndpoint = resultSet.getString("SERVER_ID");
```
Comment (karenc-bq) [inline src/test/java/testsuite/integration/utility/ContainerHelper.java:264]: I think `SERVER_ID` can be a static final string

**Pull request merged**